src/menus/serverCreationScreen.cpp

```diff
@@ -26,58 +26,75 @@ ServerSetupScreen::ServerSetupScreen()
-    server_port = new GuiTextEntry(row, "SERVER_PORT", string(defaultServerPort));
-    server_port->setSize(GuiElement::GuiSizeMax, GuiElement::GuiSizeMax);
-
-    (new GuiLabel(main_panel, "GENERAL_LABEL", tr("Server info"), 30))->addBackground()->setSize(GuiElement::GuiSizeMax, 50);
-    // Server IP row.
-    row = new GuiElement(main_panel, "");
-    row->setSize(GuiElement::GuiSizeMax, 350)->setAttribute("layout", "horizontal");
-    (new GuiLabel(row, "IP_LABEL", tr("Server IP: "), 30))->setAlignment(sp::Alignment::TopRight)->setSize(250, GuiElement::GuiSizeMax);
-    auto ips = new GuiListbox(row, "IP", [](int index, string value){});
-    ips->setSize(GuiElement::GuiSizeMax, GuiElement::GuiSizeMax);
-    ips->setTextSize(20);
-    for(auto addr_str : sp::io::network::Address::getLocalAddress().getHumanReadable())
+    
+    string reverse_proxy_value;
+    if((reverse_proxy_value = PreferencesManager::get("serverproxy")) != "")
     {
-        if (addr_str == "::1" || addr_str == "127.0.0.1") continue;
-        ips->addEntry(addr_str, addr_str);
+        (new GuiLabel(main_panel, "GENERAL_LABEL", tr("Server info"), 30))->addBackground()->setSize(GuiElement::GuiSizeMax, 50);
+        (new GuiLabel(main_panel, "SERVERPROXY_LABEL", tr("Server was launched to connect to reverse proxy:"), 30))->setSize(GuiElement::GuiSizeMax, 50);
+        string ips;
+        string sep="";
+        for(auto proxy_ip : reverse_proxy_value.split(":"))
+        {
+             ips = ips + sep + "[" + proxy_ip+ "]";
+             sep = ",";   
+        }
+        (new GuiLabel(main_panel, "SERVERPROXY_IPS", ips, 30))->setSize(GuiElement::GuiSizeMax, 50);
+    }
+    else
+    {
+        // Left column contents.
+        // General section.
+        (new GuiLabel(main_panel, "GENERAL_LABEL", tr("Server configuration"), 30))->addBackground()->setSize(GuiElement::GuiSizeMax, 50);
+    
+        // Server name row.
+        GuiElement* row = new GuiElement(main_panel, "");
+        row->setSize(GuiElement::GuiSizeMax, 50)->setAttribute("layout", "horizontal");
+        (new GuiLabel(row, "NAME_LABEL", tr("Server name: "), 30))->setAlignment(sp::Alignment::CenterRight)->setSize(250, GuiElement::GuiSizeMax);
+        server_name = new GuiTextEntry(row, "SERVER_NAME", "server");
+        server_name->setSize(GuiElement::GuiSizeMax, GuiElement::GuiSizeMax);
+
+        // Server password row.
+        row = new GuiElement(main_panel, "");
+        row->setSize(GuiElement::GuiSizeMax, 50)->setAttribute("layout", "horizontal");
+        (new GuiLabel(row, "PASSWORD_LABEL", tr("Server password: "), 30))->setAlignment(sp::Alignment::CenterRight)->setSize(250, GuiElement::GuiSizeMax);
+        server_password = new GuiTextEntry(row, "SERVER_PASSWORD", "");
+        server_password->setSize(GuiElement::GuiSizeMax, GuiElement::GuiSizeMax);
```

Comment:
If the server password isn't shown or set here, where is it shown or set? `headless_password` appears to be set only if `headless` is in use, and I don't see another way to set a server password: https://github.com/daid/EmptyEpsilon/blob/bdaa9474bffefeb82ac4c40523edc51d37f865f0/src/main.cpp#L448

Reply:
never set a server password for reverse proxy, didn't even know it could be done. It must be the same for each server ? I will check this.

Reply:
ok I guess it's using a headless server combined with a reverse proxy ? I didn't know about this headless mode but indeed some options should be usable here in reverse proxy mode